apps/dotcms-ui/src/app/portlets/shared/dot-content-types-listing/components/dot-content-type-clone-dialog/dot-content-type-clone-dialog.component.ts

```diff
@@ -0,0 +1,131 @@
+import {
+    AfterViewChecked,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    EventEmitter,
+    Input,
+    OnDestroy,
+    OnInit,
+    Output
+} from '@angular/core';
+import { DotDialogActions } from '@components/dot-dialog/dot-dialog.component';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { DotMessageService } from '@services/dot-message/dot-messages.service';
+import { DotCloneContentTypeDialogFormFields } from '@dotcms/dotcms-models';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+import { DotCMSAssetDialogCloneFields } from '@portlets/shared/dot-content-types-listing/dot-content-type.store';
+
+@Component({
+    selector: 'dot-content-type-clone-dialog',
+    templateUrl: './dot-content-type-clone-dialog.component.html',
+    styleUrls: ['./dot-content-type-clone-dialog.component.css'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DotContentTypeCloneDialogComponent implements OnInit, OnDestroy, AfterViewChecked {
+    dialogActions: DotDialogActions;
+    inputNameWithType = '';
+    @Input()
+    isVisibleDialog = false;
+    @Input()
+    isSaving = false;
+    @Output() cancelBtn = new EventEmitter<boolean>();
+
+    @Output()
+    validFormFields = new EventEmitter<DotCloneContentTypeDialogFormFields>();
+    form!: FormGroup;
+
+    asset: DotCMSAssetDialogCloneFields;
+
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+
+    constructor(
+        private readonly fb: FormBuilder,
+        private readonly dotMessageService: DotMessageService,
+        private readonly cd: ChangeDetectorRef
+    ) {
+        this.initForm();
+    }
+
+    @Input()
+    set assetSelected(assetSelected: DotCMSAssetDialogCloneFields | null) {
+        if (assetSelected) {
+            this.asset = assetSelected;
+            this.inputNameWithType = this.getNameFieldLabel(assetSelected.baseType);
+            this.patchForm(assetSelected.data);
+        }
+    }
+
+    ngOnInit(): void {
+        this.setDialogConfig();
+    }
+
+    submitForm() {
```

Comment:
we usually add doc to public methods, even if they are self explanatory

Reply:
Done.